File: src/TKDHubAPI.Domain/Entities/StudentClassAttendance.cs
namespace TKDHubAPI.Domain.Entities;
public class StudentClassAttendance : BaseEntity
{
    public int StudentClassId { get; set; }
    public StudentClass StudentClass { get; set; }

    public DateTime AttendedAt { get; set; } // Date and time of attendance

    public AttendanceStatus Status { get; set; } // e.g., Present, Absent, Late, etc.

    // Optionally, add audit fields or notes
    public string? Notes { get; set; }
}

File: Frontend/tkd_hub_web/src/app/types/StudentClassAttendance.ts
import { AttendanceStatus } from "../enums/AttendanceStatus";

export type StudentClassAttendance = {
    id: number;
    studentClassId: number;
    attendedAt: string; // ISO date string
    status: AttendanceStatus;
    notes?: string | null;
    // Optionally add: studentClass?: StudentClass;
};
